### PR TITLE
Add Sphinx docs hosted on GitHub Pages

### LINKED ISSUE #23: objects in docstrings
For linking purposes in Sphinx-created docs, it might be useful to have lines like:

https://github.com/RadioAstronomySoftwareGroup/pyradiosky/blob/9f85fee27f92c34818298570b7bf85e8bd47fc24/pyradiosky/skymodel.py#L142

use sphinx directives instead, so `:class:`astropy.EarthLocation` object`.

### PR DESCRIPTION
**Pull request by bhazelton**
---
name: Documentation
about: Update or improve the documentation.
labels: documentation
---

## Description

Add Sphinx docs hosted on ReadTheDocs (https://pyradiosky.readthedocs.io/en/latest/ -- go to the `add_rtd` version to see what's being built on this branch) and a github action to build them for a CI check.

Also add some support for doc development via an `extras_require` section and update readme and the environment.yaml, both for the docs changes and for `lunarsky` now being on PyPI.

## Motivation and Context

We need to have API docs.

closes #23
partially addresses #22

## Checklist:

- [x] I have read the [contribution guide](https://github.com/RadioAstronomySoftwareGroup/pyuvdata/blob/master/.github/CONTRIBUTING.md).
- [x] My code follows the code style of this project.
Documentation change checklist:
- [x] Any updated docstrings use the [numpy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,63 +18,75 @@ issues in the [issue log](https://github.com/RadioAstronomySoftwareGroup/pyradio
...
-[healpix, dev]. The `healpix` enables various specific functionality
-while `dev` will install dependencies required for testing and development.
 
 ## Tests
 Uses the `pytest` package to execute test suite.
 From the source pyradiosky directory run ```pytest``` or ```python -m pytest```.
 
-# Maintainers
+## Maintainers
 pyuvdata is maintained by the RASG Managers, which currently include:
```
Comment (aelanman) [inline README.md:88]: Should this be "pyradiosky is maintained by the RASG Managers"?

File: docs/make.bat [hunk 2]
```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
```
Comment (aelanman) [inline docs/make.bat:1]: Is this to support Windows?

File: pyradiosky/skymodel.py [hunk 3]
```diff
@@ -654,7 +660,7 @@ def update_positions(self, time, telescope_location):
 
         if not isinstance(telescope_location, (EarthLocation, MoonLocation)):
 
-            errm = "telescope_location must be an astropy EarthLocation object"
+            errm = "telescope_location must be an :class:`astropy.EarthLocation` object"
             if hasmoon:
                 errm += " or a lunarsky MoonLocation object "
```
Comment (aelanman) [inline pyradiosky/skymodel.py:665]: This can be replaced with `:class:lunarsky.MoonLocation`.

File: README.md (hunk 1 above)
Comment (bhazelton) [reply README.md:89]: yes, thank you!

File: docs/make.bat (hunk 2 above)
Comment (bhazelton) [reply docs/make.bat:1]: I don't know, it's autogenerated by sphinx

File: pyradiosky/skymodel.py (hunk 3 above)
Comment (bhazelton) [reply pyradiosky/skymodel.py]: updated.

File: ci/docs.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+  - scipy
+  - coverage
+  - pypandoc
+  - pytest-cov
+  - setuptools_scm
+  - sphinx
+  - pyuvdata>=2.0.1
+  - pip
+  - pip:
+      - git+https://github.com/aelanman/lunarsky.git
```
Comment (aelanman) [inline ci/docs.yaml:20]: `lunarsky` is now on pypi, if you want to change the installation path.

Comment (bhazelton) [reply ci/docs.yaml:20]: ok, I'll fix that.

**Comment by bhazelton** [comment]
`_set_spectral_type_params` is hidden because it is not something that should be done by users, only by the code. It is the method that makes various parameters required or not depending on spectral type.

**Comment by bhazelton** [comment]
I think deprecated functions should appear in docs as long as it's clear they're deprecated.

File: ci/docs.yaml (hunk 4 above)
Comment (bhazelton) [reply ci/docs.yaml]: I actually went further and added it in `extras_require` and the readme to highlight it a bit better.

**Pull request merged**